src/NetSparkle.Tools.AppCastGenerator/Program.cs

```diff
@@ -7,134 +7,215 @@
+            public string SourceBinaryDirectory { get; set; }
+
+            //[Option('g', "github-atom-feed", SetName = "github", Required = false, HelpText = "Generate from Github release atom feed (signatures not supported yet)")]
+            //public string GithubAtomFeed { get; set; }
+
+            [Option('f', "file-extract-version", SetName = "local", Required = false, HelpText = "Determine the version from the file name", Default = false)]
+            public bool FileExtractVersion { get; set; }
+
+            [Option('o', "os", Required = false, HelpText = "Operating System (windows,macos,linux)", Default = "windows")]
+            public string OperatingSystem { get; set; }
+
+            [Option('u', "base-url", SetName = "local", Required = false, HelpText = "Base URL for downloads", Default = null)]
+            public Uri BaseUrl { get; set; }
+
+            [Option('l', "change-log-url", SetName = "local", Required = false, HelpText = "File path to Markdown changelog files (expected extension: .md; version must match AssemblyVersion).", Default = "")]
+            public string ChagenLogUrl { get; set; }
+
+            [Option('p', "change-log-path", SetName = "local", Required = false, HelpText = "", Default = "")]
+            public string ChangeLogPath { get; set; }
+
+            [Option('n', "product-name", Required = false, HelpText = "Product Name", Default = "Application")]
+            public string ProductName { get; set; }
+
+            [Option('x', "url-prefix-version", SetName = "local", Required = false, HelpText = "Add the version as a prefix to the download url")]
+            public bool PrefixVersion { get; set; }
+
+
+            #region key generation
+
+            [Option("generate-keys", SetName = "keys", Required = false, HelpText = "Generate keys")]
+            public bool GenerateKeys { get; set; }
+
+            [Option("force", SetName = "keys", Required = false, HelpText = "Force regeneration of keys")]
+            public bool ForceRegneration { get; set; }
+
+            #endregion
 
-        private static string _privateKeyFilePath = "";
-        private static string _searchPattern = "*.exe";
-        private static string _baseUrl = "";
 
-        private static string _changelogsDir = "";
-        private static string _changelogsUrl = "";
+            #region signing
 
-        private static string _outputFilePath = "";
+            [Option("generate-signature", SetName = "signing", Required = false, HelpText = "Generate signature from binary")]
+            public string BinaryToSign { get; set; }
+
+            #endregion
+
+            #region
+
+            [Option("verify", SetName = "verify", Required = false, HelpText = "Binary to verify")]
+            public string BinaryToVerify { get; set; }
+
+            [Option("signature", SetName = "verify", Required = false, HelpText = "Signature")]
+            public string Signature { get; set; }
+
```

Comment:
More options I would like to see:

* [ ] Ability to pass in public key (base 64 string) for verification
* [ ] Ability to pass in private key (base 64 string) for generating signature
* [ ] Ability to output public/private key to console (we generate it, but then it can be hard to grab for CI/CD)
* [ ] Ability to generate keys and only output to console instead of to file

The above will make things a lot easier on the end user.

Reply:
**Ability to pass in private key (base 64 string) for generating signature**

This would be considered a security risk technically, the private key would now be in the shell history, persisted to the disk. (like .bash_history).  Ideally you would prompt for the user to paste it.

Reply:
Good point. I'm OK with prompting them. I think that will still be good with CI/CD since you should just be able to pipe in the key.

Reply:
I think maybe environment variables for pipeline?

Reply:
Environment variables should work too. Let's do that (environment variables) and can consider other options down the road if anyone complains.

Thank.